src/v/cluster/partition.cc

```diff
@@ -175,7 +175,7 @@ raft::replicate_stages partition::replicate_in_stages(
         }
     }
 
-    if (bid.is_transactional || bid.has_idempotent()) {
+    if (_rm_stm) {
```

Comment:
is this change  releated with commit message ?

Reply:
yes, when a request isn't transactional or idempotent we used to bypass `rm_stm` and call replicate on the consensus object directly. with this commit I want to `sync` even such regular requests so we can't bypass `rm_stm` anymore